src/emulator/controller.ts

```diff
@@ -108,6 +112,39 @@ export async function startEmulator(instance: EmulatorInstance): Promise<void> {
   await EmulatorRegistry.start(instance);
 }
 
+/**
+ * Exports emulator data on clean exit (SIGINT or process end)
+ * @param options
+ */
+export async function exportOnExit(options: any) {
+  const exportOnExitDir = options.exportOnExit;
+  if (exportOnExitDir) {
+    try {
+      options.force = true;
```

Comment:
Why are we mutating `force` and `noninteractive` here?

Reply:
Else you get a prompt that the directory already exists and if you want to override it. I'm not sure when `noninteractive` is set alternatively. I just took it from the original export logic.

Reply:
Ah I see.  Ok rather than changing these flags I think we actually want to change the logic in the export routine.  I think we should assume that if you're using `--export-on-exit` you don't need to be prompted about overwrite, since I imagine overwriting the same directory many times will be the prime use case.

So we have this check:
```
if (existingMetadata && !options.force) {
```

We should change it to something like this:
```
if (existingMetadata && !(options.force || options.exportOnExit)) {
```

`options.force` is a flag used throughout the codebase to do destructive actions, so we can't assume it's safe for us to ever modify it.

Reply:
Agreed and updated